hs_core/views/utils.py

```diff
@@ -799,6 +888,13 @@ def irods_path_is_allowed(path):
         raise SuspiciousFileOperation("File paths cannot contain '/./'")
 
 
+def irods_path_is_directory(istorage, path):
+    """ return True if the path is a directory. """
+    folder, base = os.path.split(path)
+    listing = istorage.listdir(folder)
+    return base in listing[0]
```

Comment:
This function appears to assume ```path``` input parameter should not contain trailing slash. I think it is better to accommodate trailing slash case in this function.